tests/unit/RoleCredentials.js

```diff
@@ -144,20 +144,66 @@ describe('Credentials Manager', () => {
         });
     });
 
-    it('should handle non arsenal errors on refresh', function test(done) {
-        this.timeout(10000);
-        const retryTimeout = (roleCredentials.expiration - Date.now()) +
-            1000;
-        return setTimeout(() => {
-            simulateServerError = true;
-            roleCredentials.get(err => {
-                assert(err);
-                // check that err is another instance distinct from
-                // the global arsenal error
-                assert.notStrictEqual(err, errors.InternalError);
-                done();
+    it('should properly handle Arsenal and non-Arsenal errors', done => {
+        const testCases = [
+            {
+                name: 'non-Arsenal error',
+                error: Object.assign(new Error('custom error'), {
+                    code: 'CustomError',
+                    description: 'test description'
+                }),
+                statusCode: 400,
+                expectedResult: error => {
+                    assert(error instanceof ArsenalError);
+                    assert.strictEqual(error.code, 400);
+                    assert.strictEqual(error.description, 'test description');
+                }
+            },
+            {
+                name: 'retryable Internal error',
+                error: Object.assign(new Error('internal error'), {
+                    code: 'InternalError',
+                    InternalError: true
+                }),
+                statusCode: 500,
+                expectedResult: error => {
+                    assert(error.retryable === true);
+                }
+            }
```

Comment:
should also have the case of a "real" arsenal error:

```suggestion
                error: errors.InternalError.customizeError("my internal error"),
```